src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -184,24 +256,76 @@ public override bool TryRemove(SignatureProvider signatureProvider)
                 return false;
 
             var cacheKey = GetCacheKey(signatureProvider);
+#if NET461 || NET472 || NETSTANDARD2_0 
+            MemoryCache signatureProviderCache;
+#elif NET45
+            MaxCapacityRefreshCache<string, SignatureProvider> signatureProviderCache;
+#endif
+            // Determine if we are caching a signing or a verifying SignatureProvider.
             if (signatureProvider.WillCreateSignatures)
+                signatureProviderCache = _signingSignatureProviders;
+            else
+                signatureProviderCache = _verifyingSignatureProviders;
+
+            try
             {
-                if (_signingSignatureProviders.TryRemove(cacheKey, out SignatureProvider provider))
+#if NET45
+                if (signatureProviderCache.TryRemove(cacheKey, out SignatureProvider provider))
                 {
                     provider.CryptoProviderCache = null;
                     return true;
                 }
-            }
-            else
-            {
-                if (_verifyingSignatureProviders.TryRemove(cacheKey, out SignatureProvider provider))
+                else
+                {
+                    return false; 
+                }
+#elif NET461 || NET472 || NETSTANDARD2_0
+                if (signatureProviderCache.TryGetValue(cacheKey, out SignatureProvider provider))
                 {
+                    signatureProviderCache.Remove(cacheKey);
                     provider.CryptoProviderCache = null;
                     return true;
                 }
+                else
+                {
+                    return false;
+                }
+#endif
+            }
+            catch
```

Comment:
we at least need the exception for diagnostics right? What exceptions can be thrown in the try block?

Reply:
+1, would be nice if logged when an exception is caught.

Reply:
I'm honestly not sure. I believe the only method in this try block that would throw an exception is [Remove](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.memorycache.remove), but the documentation doesn't provide any information on this. :/

I've added a log for this scenario.